src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1198,17 +1214,20 @@ def dbapi(
             set_api_call_source(res_df, DATA_SOURCE_DBAPI_SIGNATURE)
             return res_df
 
-    def _infer_data_source_schema(
-        self, conn: Connection, table: str
-    ) -> Tuple[StructType, Tuple[Tuple[str, Any, int, int, int, bool]]]:
+    def _infer_data_source_schema(self, conn: Connection, table: str) -> StructType:
         # TODO: SNOW-1893781 change implementation if oracle has different way of infer schema
-        query = f"""
-            SELECT COLUMN_NAME, DATA_TYPE, CHARACTER_MAXIMUM_LENGTH, NUMERIC_PRECISION, NUMERIC_SCALE, IS_NULLABLE
-            FROM INFORMATION_SCHEMA.COLUMNS
-            WHERE TABLE_NAME = '{table}'
-            """
-        raw_schema = conn.cursor().execute(query).fetchall()
-        return self._to_snowpark_type(raw_schema), raw_schema
+        try:
+            query = f"""
+                SELECT COLUMN_NAME, DATA_TYPE, CHARACTER_MAXIMUM_LENGTH, NUMERIC_PRECISION, NUMERIC_SCALE, IS_NULLABLE
+                FROM INFORMATION_SCHEMA.COLUMNS
+                WHERE TABLE_NAME = '{table}'
+                """
+            raw_schema = conn.cursor().execute(query).fetchall()
+            return self._to_snowpark_type(raw_schema)
```

Comment:
we might have to use raw schema in some future external datasource we want to support, but we are good for sql server and oracle db

Reply:
we can add it back any time

Reply:
sounds good to me